presto-hive/src/test/java/io/prestosql/plugin/hive/TestOriginalFilesUtils.java

```diff
@@ -0,0 +1,85 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.io.Resources;
+import io.prestosql.orc.OrcReaderOptions;
+import io.prestosql.plugin.hive.orc.OriginalFilesUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.prestosql.plugin.hive.AcidInfo.OriginalFileInfo;
+import static io.prestosql.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.prestosql.testing.TestingConnectorSession.SESSION;
+import static org.testng.Assert.assertEquals;
+
+public class TestOriginalFilesUtils
+{
+    private String tablePath;
+    private Configuration config;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        tablePath = new File(Resources.getResource(("dummy_id_data_orc")).toURI()).getPath();
+        config = new JobConf(new Configuration(false));
+    }
+
+    @Test
+    public void testGetPrecedingRowCountSingleFile()
+    {
+        List<OriginalFileInfo> originalFileInfoList = new ArrayList<>();
+        originalFileInfoList.add(new OriginalFileInfo("000001_0", 730));
+
+        long rowCountResult = OriginalFilesUtils.getPrecedingRowCount(
+                originalFileInfoList,
+                new Path(tablePath + "/000001_0"),
+                HDFS_ENVIRONMENT,
+                SESSION.getUser(),
+                new OrcReaderOptions(),
+                config,
+                new FileFormatDataSourceStats());
+        assertEquals(rowCountResult, 0, "Original file should have 0 as the starting row count");
+    }
+
+    @Test
+    public void testGetPrecedingRowCount()
+    {
+        List<OriginalFileInfo> originalFileInfos = new ArrayList<>();
+
+        originalFileInfos.add(new OriginalFileInfo("000002_0", 741));
+        originalFileInfos.add(new OriginalFileInfo("000002_0_copy_1", 768));
+        originalFileInfos.add(new OriginalFileInfo("000002_0_copy_2", 743));
+
+        long rowCountResult = OriginalFilesUtils.getPrecedingRowCount(
+                originalFileInfos,
+                new Path(tablePath + "/000002_0_copy_2"),
+                HDFS_ENVIRONMENT,
+                SESSION.getUser(),
+                new OrcReaderOptions(),
+                config,
+                new FileFormatDataSourceStats());
+        // Bucket-2 has original files: 000002_0, 000002_0_copy_1. Each file original file has 4 rows.
```

Comment:
can we have `1`, `3` and `5` rows? This would make assertion more bulletproof.

Reply:
I can, but need to generate files again from Hive, not sure it is worth the effort as we already checking 4+4.